### ISSUE TITLE
Session Globals vs Thread Globals

### ISSUE THREAD

**Issue by ShaniRotemIl** [k2view-academy/K2View-Academy#1163]
Hi,
What is the difference between globals set using setThreadGlobal function and globals set by SET command?
Is there a place today in Fabric implementation that we might want to use thread globals and not session global?

Thanks

**Comment by taliei** [comment]
Hi.

The getThreadGlobals/setThreadGlobals methods get/set the Globals in the Thread level and the set <key name> command impacts the Fabric session. The getThreadGlobals/setThreadGlobals methods are kept for compatibility reasons. 
It is recommended to use the session-level key in your implementation.

Tali

**Comment by ShaniRotemIl** [comment]
Hi Tali,

See below two tests we ran to determine when it might be better to use a thread global instead of a session global.

1. We created a BW flow that sequentially execute 5 GETs. 
Inside the LUI, we invoked getThreadGlobal for GLOBAL_TEST and logged the results. 
In the third instance, we also invoked setThreadGlobal with GLOBAL_TEST=200.
Results:
In the logs, getThreadGlobal always returned null, except for the third GET, where it printed 200.

We repeated the test, this time using a session global instead of a thread global. In this case, all the logs after the third GET printed 200.
Conclusion:
If we want a global value to affect only a specific GET, we should use a thread global. Using a session global will affect all GETs within that session.

2. In another test, we created a flow that invoked innerFlowAsync with parallel=5. The innerFlowSync executed a flow that:
- Read the session global.
- Slept for 2 seconds.
- Read the session global again.
- Logged the global values.

In the third iteration, the inner flow also set the session global to 200.
Result: All inner flows after the third iteration logged the value 200.

We repeated the test, this time using a thread global instead of a session global, but the result was the same.

Doesn’t innerFlowAsync run in a different thread?

Thanks
